### PR TITLE
[front] feature: Use dust-app for parsing table headers

### PR DESCRIPTION
**Pull request by tdraier**
## Description

Followup #7077 

Last commit restores the previous behaviour - i.e. skipping tables if headers are not "sanitizable" (too many headers with the same name, for example). The sanitization is done in front when trying to upsert a table - it will now throw a 400 with type if headers are not valid (before enqueing the upsert). 
This 400 error is already catched in microsoft, adding a catch in google drive to also skip files in that case.

The  error "Failed to parse CSV: Failed to generate unique slugified name for header \"xxx\" after multiple attempts." will now be logged as warn in connectors.

## Risk

same as #7077 

## Deploy Plan

deploy `front` (new optional parameter in api), then deploy `connectors`

### CODE REVIEW

**Comment by spolu** [comment]
>  it will now throw a 400
By throw you mean return?

File: connectors/src/connectors/notion/lib/notion_api.ts [hunk 1]
```diff
@@ -1027,7 +1023,7 @@ export async function renderDatabaseFromPages({
     )
   );
 
-  const sanitizedHeaders = getSanitizedHeaders(header);
+  const sanitizedHeaders = header;
```
Comment (spolu) [inline connectors/src/connectors/notion/lib/notion_api.ts:1026]: let's clean that up a bit

File: connectors/src/connectors/microsoft/temporal/spreadsheets.ts [hunk 2]
```diff
@@ -181,7 +182,7 @@ async function processSheet({
 
   // Assuming the first line as headers, at least one additional data line is required.
   if (rawHeaders && rows.length > 1) {
-    const headers = getSanitizedHeaders(rawHeaders);
+    const headers = rawHeaders;
```
Comment (spolu) [inline connectors/src/connectors/microsoft/temporal/spreadsheets.ts:185]: we likely don't need a new var

File: front/components/data_source/DocumentOrTableUploadOrEditModal.tsx [hunk 3]
```diff
@@ -478,6 +481,29 @@ export function DocumentOrTableUploadOrEditModal({
                 )}
               </div>
 
+              {isTable &&
+                owner.flags.includes("use_app_for_header_detection") && (
```
Comment (spolu) [inline front/components/data_source/DocumentOrTableUploadOrEditModal.tsx:485]: So for now this is the only path that triggers a header detection?

Comment (spolu) [reply front/components/data_source/DocumentOrTableUploadOrEditModal.tsx:485]: connectors upserts will never use it for now if I understand correctly?

File: front/lib/api/tables.ts [hunk 4]
```diff
@@ -297,75 +316,50 @@ export async function rowsFromCsv(
...
+
+  if (useAppForHeaderDetection) {
+    // Enable static header detection for debugging
+    const headerResStatic = await detectHeaders(auth, csv, delimiter, false);
+    logger.info(
+      { headerRes, headerResStatic, useAppForHeaderDetection },
+      "Header detection result"
+    );
+  } else {
+    logger.info(
```
Comment (spolu) [inline front/lib/api/tables.ts:334]: This log leaks user data and is kinda big. Not great

**Review by spolu — COMMENTED** [review]
Left some comments and questions

**Comment by tdraier** [comment]
> > it will now throw a 400
> By throw you mean return?

Yes, sorry , the api will return a 400 with "invalid_rows_request_error" type - and upsertTableFromCsv from connectors will throw an InvalidRowsRequestError exception when getting this 400.

File: front/components/data_source/DocumentOrTableUploadOrEditModal.tsx (hunk 3 above)
Comment (tdraier) [reply front/components/data_source/DocumentOrTableUploadOrEditModal.tsx:485]: Connector upserts for microsoft and google_drive uses header detection, when calling `upsertTableFromCsv()` - in both `spreadsheets.ts` , we have `useAppForHeaderDetection: true`. Not in notion, as databases are already well structured.

File: front/lib/api/tables.ts (hunk 4 above)
Comment (tdraier) [reply front/lib/api/tables.ts]: right, removing logs here

File: front/components/data_source/DocumentOrTableUploadOrEditModal.tsx (hunk 3 above)
Comment (spolu) [reply front/components/data_source/DocumentOrTableUploadOrEditModal.tsx:485]: For all workspaces?

File: connectors/src/lib/data_sources.ts [hunk 5]
```diff
@@ -599,6 +601,7 @@ export async function upsertTableFromCsv({
     tableId,
     truncate,
     async: true,
+    useAppForHeaderDetection,
```
Comment (spolu) [inline connectors/src/lib/data_sources.ts:604]: I don't understand this one as I don't see a diff on any v1 endpoint?

Comment (tdraier) [reply connectors/src/lib/data_sources.ts:604]: The v1 api actually calls the handler from private api ( from `front/pages/api/w/[wId]/data_sources/[dsId]/tables/csv.ts` ) and is undocumented ("System API key only endpoint. Undocumented.")

Comment (spolu) [reply connectors/src/lib/data_sources.ts:604]: It certainly does not as per the endpoint URL above

File: front/components/data_source/DocumentOrTableUploadOrEditModal.tsx (hunk 3 above)
Comment (tdraier) [reply front/components/data_source/DocumentOrTableUploadOrEditModal.tsx:485]: Only if feature flag is enabled. This is directly checked in api endpoint `front/pages/api/w/[wId]/data_sources/[dsId]/tables/csv.ts`, you must have the parameter set and the flag.

File: connectors/src/lib/data_sources.ts (hunk 5 above)
Comment (spolu) [reply connectors/src/lib/data_sources.ts:604]: As per IRL the handler is shared

File: front/components/data_source/DocumentOrTableUploadOrEditModal.tsx (hunk 3 above)
Comment (spolu) [reply front/components/data_source/DocumentOrTableUploadOrEditModal.tsx:485]: ack

**Review by spolu — APPROVED** [review]
LGTM

Functionally LGTM. Still a bit of FUD on how to properly rolll that out to a workspace that is not ours.

Probably worth spending a bit of time writing up short document on that after this starts being tested on our workspace so that it can trigger a discussion.

(cost evaluation based on real data, possible impact of enabling it, etc...)

**Pull request merged**